service/controller/azuremachinepool/handler/nodepool/create.go

```diff
@@ -89,11 +101,40 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func isMasterUpgrading(getter key.AnnotationsGetter) bool {
-	masterUpgrading, exists := getter.GetAnnotations()[annotation.IsMasterUpgrading]
-	if !exists {
-		return false
+func (r *Resource) isMasterUpgrading(ctx context.Context, amp *v1alpha3.AzureMachinePool) (bool, error) {
+	r.Logger.Debugf(ctx, "Checking if master nodes are upgrading")
+
+	cluster, err := capiutil.GetClusterFromMetadata(ctx, r.CtrlClient, amp.ObjectMeta)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	// get ctrl client on the WC
+	wcClient, err := r.tenantClientFactory.GetClient(ctx, cluster)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	// Get a list of all nodes, including worker nodes.
+	nodeList := v1.NodeList{}
+	err = wcClient.List(ctx, &nodeList)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	for _, node := range nodeList.Items {
+		if node.Labels["role"] == "master" || node.Labels["kubernetes.io/role"] == "master" {
+			// Check if node has the right azure operator version label
```

Comment:
so we no longer rely on `annotation.IsMasterUpgrading` being set?

Reply:
It was not set anywhere.